crypto/ec/ec_key.c

```diff
@@ -350,6 +351,43 @@ static int ec_generate_key(EC_KEY *eckey, int pairwise_test)
     return ok;
 }
 
+#ifndef FIPS_MODULE
+/*
+ * This is similar to ec_generate_key(), except it uses an ikm to
+ * derive the private key.
+ */
+int ossl_ec_generate_key_dhkem(EC_KEY *eckey,
+                               const unsigned char *ikm, size_t ikmlen)
+{
+    int ok = 0;
+
+    if (eckey->priv_key == NULL) {
+        eckey->priv_key = BN_secure_new();
+        if (eckey->priv_key == NULL)
+            goto err;
+    }
+    if (ossl_ec_dhkem_derive_private(eckey, eckey->priv_key, ikm, ikmlen) <= 0)
+        goto err;
+    if (eckey->pub_key == NULL) {
+        eckey->pub_key = EC_POINT_new(eckey->group);
+        if (eckey->pub_key == NULL)
+            goto err;
+    }
+    if (!ossl_ec_key_simple_generate_public_key(eckey))
```

Comment:
Probably this should be `eckey->group->meth->keygenpub(eckey)`.

Reply:
I  can now see where you are going with this, however this code is a variation of ec_generate_key().
If this is wrong then so is ossl_ec_key_pairwise_check(). So this to me looks more like a refactor task that is out of scope for this PR.

Reply:
@mattcaswell any additional input here?

Reply:
If it had of just fit into the existing keygen as an extra 'if else clause' there would probably not have been a question related to this :)